### PR TITLE
fix(aws_instance_profile): split profiles per node type

### PR DESCRIPTION
**Pull request by fruch**
Split the configuration, so we can select specific profile for each type of node, and not use the same one for all of the nodes.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: defaults/aws_config.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ ami_db_scylla_user: 'scyllaadm'
 ami_loader_user: 'centos'
 # centos7 is used for monitor
 ami_monitor_user: 'centos'
-aws_instance_profile_name: 'qa-scylla-manager-backup-instance-profile'
+aws_instance_profile_name_db: 'qa-scylla-manager-backup-instance-profile'
```
Comment (soyacz) [inline defaults/aws_config.yaml:40]: shouldn't add also value for `aws_instance_profile_name_loader` profile?

File: sdcm/sct_config.py [hunk 2]
```diff
@@ -672,8 +672,11 @@ class SCTConfiguration(dict):
         dict(name="extra_network_interface", env="SCT_EXTRA_NETWORK_INTERFACE", type=boolean,
              help="if true, create extra network interface on each node"),
 
-        dict(name="aws_instance_profile_name", env="SCT_AWS_INSTANCE_PROFILE_NAME", type=str,
-             help="This is the name of the instance profile to set on all instances"),
+        dict(name="aws_instance_profile_name_db", env="SCT_AWS_INSTANCE_PROFILE_NAME_DB", type=str,
+             help="This is the name of the instance profile to set on all db instances"),
+
+        dict(name="aws_instance_profile_name_loader", env="SCT_AWS_INSTANCE_PROFILE_NAME_LOADER", type=str,
+             help="This is the name of the instance profile to set on all db instances"),
```
Comment (soyacz) [inline sdcm/sct_config.py:679]: should refer to loader

File: sdcm/cluster_aws.py [hunk 3]
```diff
@@ -121,6 +121,14 @@ def __str__(self):
     def calculate_spot_duration_for_test(self):
         return floor(self.test_config.TEST_DURATION / 60) * 60 + 60
 
+    @property
+    def instance_profile_name(self) -> str | None:
+        if self.node_type == "scylla-db":
+            return self.params.get('aws_instance_profile_name_db')
+        if self.node_type == "loader":
+            return self.params.get('aws_instance_profile_name_loader')
+        return None
```
Comment (soyacz) [inline sdcm/cluster_aws.py:130]: what about monitor?
Shouldn't be there some default for other cases?

File: defaults/aws_config.yaml (hunk 1 above)
Comment (fruch) [reply defaults/aws_config.yaml:40]: no, since by default we don't add to it.

File: sdcm/cluster_aws.py (hunk 3 above)
Comment (fruch) [reply sdcm/cluster_aws.py:130]: we don't have a use case yet that we need it for monitor.

when we will, we'll add this configuration and code to handle it.

File: sdcm/sct_config.py (hunk 2 above)
Comment (fruch) [reply sdcm/sct_config.py]: fixed

**Review by soyacz — APPROVED** [review]

**Review by roydahan — APPROVED** [review]

**Pull request merged**